source/beamline/PGM/PGMBPMControl.cpp

```diff
@@ -0,0 +1,41 @@
+#include "beamline/PGM/PGMBPMControl.h"
+
+PGMBPMControl::PGMBPMControl(const QString &name, const QString &pvName, int avgValue, int variance, QObject *parent, const QString description)
+	: AMReadOnlyPVControl(name, pvName, parent, description)
+
+{
+    averageValue_ = avgValue;
+    variance_ = variance;
+	validityState_ = true;
+
+	// Get updated value from AMReadOnlyPVControl.
+	connect(this, SIGNAL(valueChanged(double)), this, SLOT(onPVValueChanged(double)));
+
+}
+
+QString PGMBPMControl::valueStr() const
```

Comment:
In fact I'm wondering if we need to do the string conversion here at all. Shouldn't it be up to the receiver of the usual valueChanged() signal to do the conversion if they wish to?